common/djangoapps/edraak_social/forus/pipeline.py

```diff
@@ -0,0 +1,67 @@
+from social_core.pipeline.social_auth import associate_by_email
+from social_core.pipeline.user import create_user
+
+from student.models import UserProfile
+
+from .utils import forus_authentication_only
+
+
+@forus_authentication_only
+def save_redirect_param(strategy, *args, **kwargs):
+    strategy.request.session['next'] = \
+        strategy.request.GET.get('next', '').replace(' ', '+')
```

Comment:
We might need to ask Forus to escape the parameter using a function like `next = encodeURIComponent(next)` or `HttpUtility.UrlEncode` in C#

```suggestion
        strategy.request.GET.get('next', '')
```

Reply:
After reviewing the authentication implementation as well as the steps followed by Forus for generating the next parameter, I can confirm that the next parameters of incoming requests are URI encoded before being sent to Edraak. However, once these the next parameter is received and decoded the '+' character is mistaken for a space character. This step replaces spaces with + signs once again.